xarray/core/dataset.py

```diff
@@ -1125,27 +1200,25 @@ def isel(self, drop=False, **indexers):
         See Also
         --------
         Dataset.sel
-        Dataset.sel_points
-        Dataset.isel_points
         DataArray.isel
         """
-        invalid = [k for k in indexers if k not in self.dims]
-        if invalid:
-            raise ValueError("dimensions %r do not exist" % invalid)
+        indexers_list = self._validate_indexers(indexers)
 
-        # all indexers should be int, slice or np.ndarrays
-        indexers = [(k, (np.asarray(v)
-                         if not isinstance(v, integer_types + (slice,))
-                         else v))
-                    for k, v in iteritems(indexers)]
+        coord_vars = self._get_indexers_coordinates(indexers)
+        coord_names = set(self._coord_names) | set(coord_vars)
 
         variables = OrderedDict()
         for name, var in iteritems(self._variables):
-            var_indexers = dict((k, v) for k, v in indexers if k in var.dims)
+            var_indexers = {k: v for k, v in indexers_list if k in var.dims}
             new_var = var.isel(**var_indexers)
             if not (drop and name in var_indexers):
                 variables[name] = new_var
-        coord_names = set(self._coord_names) & set(variables)
+
+        # attatch / overwrite coordinate in indexers
+        for k, v in coord_vars.items():
+            variables[k] = v
```

Comment:
`variables.update(coord_vars)` does this without the loop

Reply:
Done